devstack/plugin.sh

```diff
@@ -223,15 +223,34 @@ elif [[ "$1" == "stack" && "$2" == "pre-install" ]]; then
         # Packages should have been installed by devstack
         #install_package $(_parse_package_files $CONTRAIL_DEST)
 
+        # From R4.0, IFMAP server (i.e. irond) dependency was removed
+        if _vercmp $CONTRAIL_BRANCH ">=" R4.0 && is_package_installed ifmap-server; then
+            uninstall_package ifmap-server || true #ifmap-server uninstall does not exit properly
+            # Some directories was install by the ifmap-server package
+            sudo mkdir -p /var/lib/contrail
+            sudo chown -R $STACK_USER. /var/lib/contrail/
+            sudo mkdir -p /var/log/contrail
+            sudo chown -R $STACK_USER:adm /var/log/contrail
+            sudo chmod 0750 /var/log/contrail
+            sudo chown -R $STACK_USER. /etc/contrail
+        fi
+
         echo_summary "Building contrail"
         cd $CONTRAIL_DEST
         sudo -E scons $SCONS_ARGS
         cd $TOP_DIR
 
         # As contrail's python packages requirements aren't installed
         # automatically, we have to manage their installation.
-        pip_install -r $CONTRAIL_PLUGIN_DIR/files/requirements.txt
-
+        if _vercmp $CONTRAIL_BRANCH ">=" R4.0; then
+            sed '/^discoveryclient$/d' $CONTRAIL_PLUGIN_DIR/files/requirements.txt > /tmp/contrail_requirements.txt
+            pip_install -r /tmp/contrail_requirements.txt
+        else
+            pip_install -r $CONTRAIL_PLUGIN_DIR/files/requirements.txt
+        fi
+        # Force to use the version 0.9.3 of the Thrift python library as it is a requirements for Pycassa library.
+        # Recent OpenStack release require at least Thrift 0.10.0.
+        sudo pip install -U thrift==0.9.3
```

Comment:
This was (maybe partially) adressed by commit [0989d1eb76e9e151932e45d4fdc913991bcb8e3](https://github.com/zioc/contrail-devstack-plugin/commit/e0989d1eb76e9e151932e45d4fdc913991bcb8e3)

Wouldn't it be better to put thrift==0.9.3 in requirements.txt ?